rasa/utils/tensorflow/models.py

```diff
@@ -211,11 +228,33 @@ def fit(
                         )
 
                     val_results = self._get_metric_results(prefix="val_")
+                    self._save_model_checkpoint(current_results=val_results, epoch=epoch)
 
                 postfix_dict.update(val_results)
 
             progress_bar.set_postfix(postfix_dict)
 
+        # Checkpoint the model one last time after training
+        if evaluate_on_num_examples > 0 and self.checkpoint_model:
+            epoch_batch_size = self.linearly_increasing_batch_size(
+                epochs, batch_size, epochs
+            )
+            self._batch_loop(
+                evaluation_dataset_function,
+                tf_evaluation_on_batch_function,
+                epoch_batch_size,
+                False,
+                training_steps,
+                self.test_summary_writer,
+            )
+
+            val_results = self._get_metric_results(prefix="val_")
+            self._save_model_checkpoint(current_results=val_results, epoch=epochs)
```

Comment:
If you take a look at 
```
    def _should_evaluate(
        evaluate_every_num_epochs: int, epochs: int, current_epoch: int
    ) -> bool:
        return (
            current_epoch == 0
            or (current_epoch + 1) % evaluate_every_num_epochs == 0
            or (current_epoch + 1) == epochs
        )
```
you see that we always evaluate also on the last epoch. So I think, we don't need this code here.

Reply:
just double checked (tried it out locally), we already evaluate after the last epoch, so we don't need this